### PR TITLE
Lightning text

### PR DESCRIPTION
**Pull request by jdnurme**
Add demo code for running text-based parquet model training with pytorch lightning and dataflux.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: demo/lightning/text-based/model.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                for batch in table.iter_batches(batch_size=self.batch_size,
+                                                columns=self.columns):
+                    yield from batch.to_pylist()
+
+
+def main():
+    args = parse_args()
+    logging.basicConfig(level=args.log_level)
+    # The prefix passed here determines which parquet files are read from our bucket.
+    # config = df_iter.Config(prefix="fineweb/sample/10BT/014_00000.parquet")
```
Comment (MattIrv) [inline demo/lightning/text-based/model.py:194]: Did you mean to leave this commented out?

File: demo/lightning/text-based/model.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+from torch.utils import data
+from torch.nn.functional import pad
+from transformers import AutoTokenizer
+
+# This demo was built around the huggingface fineweb dataset.
+# Details of the dataset can be found at https://huggingface.co/datasets/HuggingFaceFW/fineweb/tree/main/sample/10BT
+
+# Example execution:
+# python3 ./model.py --project=test-project --bucket=my-fineweb-data --batch-size=9 --prefix="fineweb/sample/10BT"
+
```
Comment (MattIrv) [inline demo/lightning/text-based/model.py:37]: In the comment here, could you add a note about the expected performance of this demo / has it been optimized or benchmarked for performance at all?

**Review by MattIrv — APPROVED** [review]

File: demo/lightning/text-based/model.py (hunk 1 above)
Comment (jdnurme) [reply demo/lightning/text-based/model.py:194]: Nope, will remove

File: demo/lightning/text-based/model.py (hunk 2 above)
Comment (jdnurme) [reply demo/lightning/text-based/model.py:37]: This hasn't been benched at all, I was going to add a timer and some description of that in the next PR after some more thorough benchmarking.

Comment (MattIrv) [reply demo/lightning/text-based/model.py:37]: For now I would recommend adding a sentence as a caveat saying that it's for simple demo purposes only and has not been optimized for performance at all

**Review by divrawal — APPROVED** [review]

**Pull request merged**